### PR TITLE
Fix #641: Use environment variables to specify which gfx targets to compile

### LINKED ISSUE #641: Provide a way to specify which gfx targets to compile for in the PKGBUILD
I have a Radeon RX 6600 XT, and while ROCm does not officially support gfx1032 yet, I was able to manually edit the `CMakeLists.txt` for the various repositories to compile for the gfx1032 target. This also decreased my compile times by a lot and the resulting binaries were much smaller since they didn't contain binaries for any other targets. For instance, rocFFT is only 600MB when only compiling for gfx1032 instead of 3GB. I'm wondering if a programmatic way to do this could be added to the ROCm PKGBUILDs.

### PR DESCRIPTION
**Pull request by Ta180m**
This PR adds the additional line `-DAMDGPU_TARGETS=${AMDGPU_TARGETS}` to each of the cmake commands to enable specifying the target to compile for using an environment variable, and otherwise defaults to compiling all targets. This saves a significant amount of time and binary size for people compiling on their own machines, since they can now only compile for their particular GPU target.

Resolves #641

### CODE REVIEW

**Comment by throwm8** [comment]
You can just pass -DAMDGPU_TARGETS=${AMDGPU_TARGETS} to cmake. I used this(with the variable replaced with my gpu's arch)  to speed up the builds of rocFFT and rocBLAS so it should do what you desire.

**Comment by Ta180m** [comment]
> You can just pass -DAMDGPU_TARGETS=${AMDGPU_TARGETS} to cmake. I used this(with the variable replaced with my gpu's arch) to speed up the builds of rocFFT and rocBLAS so it should do what you desire.

Thank you for the suggestion! I'll modify my PR to that instead.

**Comment by Ta180m** [comment]
The commit history is currently very messy because I've had to constantly merge the upstream master branch into mine to get the latest commits, but the maintainers can squash merge to clean it up when they merge this PR.

**Review by acxz — CHANGES_REQUESTED** [review]
I actually don't think this is a bad idea, however, I want to be a more strict with you if that's okay.

1) Can you add the link to the line of code where each project uses the `ADMGPU_TARGETS` variable in the extended commit message? This is just to make sure that upstream uses this variable and we can keep track of it if they ever change it.
2) Can you also point to what targets `-all` gets expanded to?
3) When updating PKGBUILDs please bump the pkgrel and update the `.SRCINFO`. Search these terms in the archwiki for more info
4) Also go ahead and give `squashing` yourself a try! It's not that bad and you'll learn something!

**Comment by Ta180m** [comment]
>     3. When updating PKGBUILDs please bump the pkgrel and update the `.SRCINFO`. Search these terms in the archwiki for more info

Are you sure you want me to bump the pkgrel? Wouldn't that make most people do an unnecessary rebuild?

**Review by acxz — APPROVED** [review]
Thanks for the clean up, yeah you are right we don't have to update the pkgrel.

This looks like a nice qol change for individuals compiling this PKGBUILDs. I think when we our binary stack up and running again on arch4edu then this variable should be empty, but for people compiling just for their own machine I like this.

@tpkessler I'll let you have the final say/merge on this.

**Comment by Ta180m** [comment]
@tpkessler Any other changes that you want me to make to this PR?

**Comment by tpkessler** [comment]
This is looking good @Ta180m! Could you please open a separate issue with the list of `AMDGPU_TARGETS` you added to the commit message? This makes it easier to keep track of changes by AMD.

**Comment by tpkessler** [comment]
> Can you also point to what targets -all gets expanded to?

I had to reintroduce the default targets. Otherwise the package building fails at the linking step.

**Comment by Ta180m** [comment]
> > Can you also point to what targets -all gets expanded to?
> 
> I had to reintroduce the default targets. Otherwise the package building fails at the linking step.

Which packages were you having trouble building without the default targets? I didn't have any issues.

**Comment by tpkessler** [comment]
Let's continue the discussion here https://github.com/rocm-arch/rocm-arch/discussions/695

**Pull request merged**
